lib/practice_exercises.rb

```diff
@@ -1,13 +1,61 @@
-
-# Time Complexity: ?
-# Space Complexity: ?
+require "pry"
+# Time Complexity: O(n2) because it is a nested loop.  Each loop runs n number of times where n is the 
+# length of the input test_array. Since the loops are nested, the loop will run n X n number of times.
+# Space Complexity: O(1) because the amount of memory used does not change as the size of the input changes. 
+list = [1, 2, 2, 3, 3, 4]
 def remove_duplicates(list)
-  raise NotImplementedError, "Not implemented yet"
+
+  i = 1
+  c = 0
+
+  while list[c] != nil
+    while list[c] == list[i]
```

Comment:
This is a little odd and I think you could have `i` go beyond the bounds of the array.  Something could break here.